templates/components/products/product-view.html

```diff
@@ -76,10 +76,39 @@ <h2 class="productView-brand"{{#if schema}} itemprop="brand" itemscope itemtype=
                     <dt class="productView-info-name">{{lang 'products.availability'}}</dt>
                     <dd class="productView-info-value">{{product.availability}}</dd>
                 {{/if}}
-                {{#if product.weight}}
+                {{#if theme_settings.show_product_weight}}
```

Comment:
you still need to check for `product.weight`. You can use `all` helper

```
{{#all product.weight theme_settings.show_product_weight}}
```

Reply:
Added this to weight as well as dimensions since dimensions are optional. Thanks for catching these!